File: AdventOfCode/2024/1/solution.py
import sys

left = []
right = []

for line in sys.stdin:
    line = line.rstrip()
    if line == '':
        break

    a, b = line.split()
    left.append(int(a))
    right.append(int(b))

left.sort()
right.sort()

total_diff = 0

for (a, b) in zip(left, right):
    total_diff += abs(a - b)

print(total_diff)

File: AdventOfCode/2024/1/solution.go
package main

import (
	"fmt"
	"math"
	"slices"
)

func main() {
	var a, b int
	var left, right []int

	for {
		n, _ := fmt.Scanf("%d %d\n", &a, &b)
		if n == 0 {
			break
		}

		left = append(left, a)
		right = append(right, b)
	}

	slices.Sort(left)
	slices.Sort(right)

	total_diff := 0

	for i := range left {
		total_diff += int(math.Abs(float64(left[i] - right[i])))
	}

	fmt.Println(total_diff)

	// fmt.Printf("%v", left)
	// fmt.Printf("%v", right)
}
